lib/redix/connection.ex

```diff
@@ -22,23 +22,54 @@ defmodule Redix.Connection do
     tail: "",
     # TODO: document this
     continuation: nil,
+    # TODO: document this
+    current_backoff: nil,
   }
 
+  @initial_backoff 500
+
+  @backoff_exponent 1.5
+
   ## Callbacks
 
   @doc false
   def init(opts) do
-    {:connect, :init, Map.put(@initial_state, :opts, opts)}
+    state = Map.put(@initial_state, :opts, opts)
+
+    if opts[:sync_connect] do
+      sync_connect(state)
+    else
+      {:connect, :init, state}
+    end
   end
 
   @doc false
   def connect(info, state)
 
   def connect(info, state) do
-    case Utils.connect(info, state) do
+    case Utils.connect(state) do
       {:ok, state} ->
         state = start_receiver_and_hand_socket(state)
         {:ok, state}
+      {:error, reason} ->
+        Logger.error [
+          "Failed to connect to Redis (", Utils.format_host(state), "): ",
```

Comment:
@lexmag do you think we should mention this is a *reconnection* instead of a generic *connection* attempt here? I think it may be fine like this.